### PR TITLE
HBASE-25292 [branch-1] Improve InetSocketAddress usage discipline

### PR DESCRIPTION
**Pull request by apurtell**
See description on https://github.com/apache/hbase/pull/2669

### CODE REVIEW

File: hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/BlockingRpcConnection.java [hunk 1]
```diff
@@ -253,7 +252,17 @@ protected void setupConnection() throws IOException {
         if (this.rpcClient.localAddr != null) {
           this.socket.bind(this.rpcClient.localAddr);
         }
-        NetUtils.connect(this.socket, remoteId.getAddress(), this.rpcClient.connectTO);
+        if (this.rpcClient.metrics != null) {
```
Comment (apurtell) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/BlockingRpcConnection.java:255]: This change fixes a bug we encountered in production while running in Amazon's Elastic Kubernetes Service.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcConnection.java [hunk 2]
```diff
@@ -216,24 +220,33 @@ public void operationComplete(Future<Boolean> future) throws Exception {
     });
   }
 
-  private void connect() {
+  private void connect() throws UnknownHostException {
     if (LOG.isDebugEnabled()) {
-      LOG.debug("Connecting to " + remoteId.address);
+      LOG.debug("Connecting to " + remoteId.getAddress());
```
Comment (apurtell) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcConnection.java:225]: This is the netty version of the bug fix here: https://github.com/apache/hbase/pull/2671/files#diff-1a7ec27a8107293b6c87132823c262fc250570687a40f45646c43ae46dc6b04eR255

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  docker  |   6m 38s |  Docker failed to build yetus/hbase:b249092a5f.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/2671 |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2671/1/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  docker  |   0m  7s |  Docker failed to build yetus/hbase:b249092a5f.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/2671 |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2671/2/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  docker  |   0m  9s |  Docker failed to build yetus/hbase:b249092a5f.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/2671 |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2671/3/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  docker  |   6m 28s |  Docker failed to build yetus/hbase:b249092a5f.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/2671 |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2671/4/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   8m  7s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 4 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 23s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m 40s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   2m  0s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   2m 17s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   4m  6s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   4m 19s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 51s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   2m 13s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   3m 43s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   7m 55s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 29s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 54s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   1m 54s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 14s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| -1 :x: |  javac  |   0m 56s |  hbase-server-jdkAzulSystems,Inc.-1.7.0_272-b10 with JDK Azul Systems, Inc.-1.7.0_272-b10 generated 2 new + 4 unchanged - 2 fixed = 6 total (was 6)  |
| -1 :x: |  checkstyle  |   0m 48s |  hbase-client: The patch generated 1 new + 168 unchanged - 3 fixed = 169 total (was 171)  |
| -1 :x: |  checkstyle  |   2m  7s |  hbase-server: The patch generated 1 new + 108 unchanged - 1 fixed = 109 total (was 109)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m  2s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   6m 17s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   1m 49s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   2m  6s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   6m 35s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 55s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 45s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  |  92m 31s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  unit  |   0m 33s |  hbase-external-blockcache in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 34s |  The patch does not generate ASF License warnings.  |
|  |   | 181m 32s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestConnectionImplementation |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2671/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2671 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 151ce7cf0760 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2671/out/precommit/personality/provided.sh |
| git revision | branch-1 / 30d988f |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
| javac | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2671/5/artifact/out/diff-compile-javac-hbase-server-jdkAzulSystems,Inc.-1.7.0_272-b10.txt |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2671/5/artifact/out/diff-checkstyle-hbase-client.txt |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2671/5/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2671/5/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2671/5/testReport/ |
| Max. process+thread count | 3698 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-client hbase-server hbase-external-blockcache U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2671/5/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by apurtell** [comment]
https://github.com/apache/hbase/pull/2669 was merged, closing this and picking back the master commit.

**Pull request closed without merging**